scripts/python/qa-scripts/createProposersAggregate.py

```diff
@@ -0,0 +1,145 @@
+                            ratioColVal = self.checkIfMarked(
+                                locAss, self.opt.notValidRationaleCol
+                            )
+                            if (ratioColVal > 0):
+                                self.dfMasterProposers.loc[id, self.opt.proposersRationaleCol] = locAss[self.opt.notValidRationaleCol]
+
+        self.dfMasterProposers[self.opt.assessmentsIdCol] = self.dfMasterProposers.index
+        self.dfMasterProposers.to_csv('cache/test-proposers-aggregate.csv')
+        return
+        spreadsheet = self.gspreadWrapper.createDoc(self.opt.proposersAggregateFileName)
+
+        # Print valid assessments
+        assessmentsHeadings = [
+            self.opt.assessmentsIdCol, self.opt.challengeCol,
+            self.opt.proposalKeyCol, self.opt.ideaURLCol, self.opt.assessorCol,
+            self.opt.tripletIdCol, self.opt.proposalIdCol,
+            self.opt.q0Col, self.opt.q0Rating, self.opt.q1Col, self.opt.q1Rating,
+            self.opt.q2Col, self.opt.q2Rating, self.opt.blankCol,
+            self.opt.proposerMarkCol, self.opt.proposersRationaleCol
+        ]
+        assessmentsWidths = [
+            ('A', 30), ('B:D', 150), ('E', 100), ('F:G', 40), ('H', 300),
+            ('I', 30), ('J', 300), ('K', 30), ('L', 300), ('M:0', 30),
+            ('P', 300)
+        ]
+        assessmentsFormats = [
+            ('A', self.utils.counterFormat),
+            ('I', self.utils.counterFormat),
+            ('K', self.utils.counterFormat),
+            ('M', self.utils.counterFormat),
+            ('N', self.utils.counterFormat),
+            ('A1:P1', self.utils.headingFormat),
+            ('N1:O1', self.utils.verticalHeadingFormat),
+            ('I1', self.utils.verticalHeadingFormat),
+            ('K1', self.utils.verticalHeadingFormat),
+            ('M1', self.utils.verticalHeadingFormat),
+            ('H2:H', self.utils.textFormat),
+            ('J2:J', self.utils.textFormat),
+            ('L2:L', self.utils.textFormat),
+        ]
+
+        self.gspreadWrapper.createSheetFromDf(
+            spreadsheet,
+            'Assessments',
+            self.dfMasterProposers,
+            assessmentsHeadings,
+            columnWidths=assessmentsWidths,
+            formats=assessmentsFormats
+        )
+
+        print('Proposers Aggregated Document created')
+        print('Link: {}'.format(spreadsheet.url))
+
+    def checkIntegrity(self, id, ass1, ass2):
+        if (
+            (ass1[self.opt.proposalIdCol] != ass2[self.opt.proposalIdCol]) or
+            (ass1[self.opt.q0Rating] != ass2[self.opt.q0Rating]) or
+            (ass1[self.opt.q1Rating] != ass2[self.opt.q1Rating]) or
+            (ass1[self.opt.q2Rating] != ass2[self.opt.q2Rating]) or
+            (ass1[self.opt.assessorCol] != ass2[self.opt.assessorCol])
+        ):
+            return False
+        return True
+
+    def checkIfMarked(self, row, column):
+        if (row[column].strip() != ''):
+            return 1
```

Comment:
Allowing to return an integer instead of a bool here, allows to have the final sum in the aggregated file. In case of proposers is useful to spot double submissions and for vCA is needed to calculate the consensus.

Reply:
Just a nit, but you can sum boolean values just fine, and it would make reading this function a little bit easier, as you could write
```
def checkIfMarked(self, row, column):
    row[column].strip() != ''
```